src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -267,21 +310,43 @@ void BaseCouplingScheme::secondExchange()
 void BaseCouplingScheme::storeExtrapolationData()
 {
   PRECICE_TRACE(_timeWindows);
+  // @todo breaks for CplSchemeTests/SerialImplicitCouplingSchemeTests/testConfiguredAbsConvergenceMeasureSynchronized. Why? @fsimonis
+  // for (auto &data : getAllData() | boost::adaptors::map_values) {
+  //   data->storeExtrapolationData();
+  // }
   for (auto &pair : getAllData()) {
-    PRECICE_DEBUG("Store data: {}", pair.first);
     pair.second->storeExtrapolationData();
   }
```

Comment:
I was only partially successful when trying to use `map_values`. I marked the corresponding sections. Do you see what I'm doing wrong here?

Reply:
I think I found a hint now: removing the function `getAllData()` and instead creating and maintaining another map `_cplData` allows us to use `boost::adaptors::map_values`. For `getAccelerationData()` the situation is similar, but I did not want to create another map for acceleration data. See [708617b](https://github.com/precice/precice/pull/1523/commits/708617b14a741ac1f3fb7cb74e28192311ece277).

I will also open an issue, because `getAllData()` has some more problems in connection with multi coupling.

Reply:
#1535 fixes the problems here. Like mentioned above: `getAccelerationData()` is still a problem.